addons/token-exchange/agent_mirrorpeer_controller.go

```diff
@@ -86,12 +88,68 @@ func (r *MirrorPeerReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 		klog.Error(err, "Failed to get MirrorPeer")
 		return ctrl.Result{}, err
 	}
+
 	scr, err := utils.GetCurrentStorageClusterRef(&mirrorPeer, r.SpokeClusterName)
 	if err != nil {
 		klog.Error(err, "Failed to get current storage cluster ref")
 		return ctrl.Result{}, err
 	}
 
+	if mirrorPeer.GetDeletionTimestamp().IsZero() {
+		if !utils.ContainsString(mirrorPeer.GetFinalizers(), SpokeMirrorPeerFinalizer) {
+			klog.Info("Finalizer not found on MirrorPeer. Adding Finalizer")
+			mirrorPeer.Finalizers = append(mirrorPeer.Finalizers, SpokeMirrorPeerFinalizer)
+			if err := r.HubClient.Update(ctx, &mirrorPeer); err != nil {
+				klog.Info("Failed to add finalizer to MirrorPeer")
+				return ctrl.Result{}, err
+			}
+		}
+	} else {
+		klog.Info("Mirrorpeer is being deleted")
+		peerRef, err := utils.GetCurrentMirrorPeer(&mirrorPeer, r.SpokeClusterName)
```

Comment:
We can encapsulate all this in `func uninstall`and put the function in agent_mirrorpeer_delete.go
Will make this file cleaner